src/LabLog/Services/SchoolService.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using LabLog.Models;
+using Microsoft.EntityFrameworkCore;
+using LabLog.Domain.Events;
+
+
+namespace LabLog.Services
```

Comment:
A lot of these db calls should be Async. Look for methods of the same name as you have, but with Async at the end. Eg SingleOrDefaultAsync()
This will require a few modifications to your interfaces to support properly, but the performance improvements are staggering and it will save you a lot of bugs in some APIs later

Reply:
If a synchronous method calls a method which calls an async method (with the await, etc), will the synchronous method pause and wait for the async to complete? Or will it just run on, unawares that there is an async waiting in the background?

Reply:
Async everywhere. This is what I meant by "this will require a few modifications". Dotnet core supports async all the way to the top, and you should use it. Can talk more about it if you want